clients/nodejs/cac-client/client.ts

```diff
@@ -0,0 +1,106 @@
+import * as ffi from 'ffi-napi';
+import * as ref from 'ref-napi';
+import * as path from 'path';
+import os from 'os';
+
+let platform = os.platform();
+
+let file =
+    platform == "darwin" ?
+    '../../../target/debug/libcac_client.dylib' :
+    platform == "linux" ?
+    '../../../target/debug/libcac_client.dll' :
+    '../../../target/debug/libcac_client.so';
+
+const libPath = path.join(__dirname, file);
+
+const refType = ref.types;
+const int = refType.int;
+const string = refType.CString;
+const voidType = refType.void;
+
+class CacClient {
+    tenant: string | null = null;
+    cacHostName: string | null = null;
+    pollingFrequency: number = 10;
+
+    rustLib = ffi.Library(libPath, {
+        'cac_new_client': [int, [string, int, string]],
+        'cac_get_client': [string, [string]],
+        'cac_start_polling_update': [voidType, [string]],
+        'cac_free_client': [voidType, [string]],
+        'cac_last_error_message': [string, []],
+        'cac_get_config': [string, [string, string, string]],
+        'cac_last_error_length': [int, [voidType]],
+        'cac_free_string': [voidType, [string]],
+        'cac_get_last_modified': [string, [string]],
+        'cac_get_resolved_config': [string, [string, string, string, string]],
+        'cac_get_default_config': [string, [string, string]],
+
+    });
+
+    constructor(tenantName: string, pollingFrequency: number, cacHostName: string) {
+        if (tenantName == "" || cacHostName == "") {
+            throw Error("tenantName cannot be null or empty")
+        }
+        this.tenant = tenantName;
+        this.pollingFrequency = pollingFrequency;
+        this.cacHostName = cacHostName;
+    }
+
+    public getCACLastErrorMessage(): string | null {
```

Comment:
Can we return just string?

Reply:
@Datron we cannot just return string as the response type that ref-napi libraray is providing has return type string | null